key/src/address.rs

```diff
@@ -90,10 +92,15 @@ impl FromStr for FullAddress {
     fn from_str(s: &str) -> Result<Self, Error>
     where
         Self: Sized, {
-        let decoded = Bech32::from_string(s.to_string())?;
+        if s.len() < 7 {
+            return Err(Error::Bech32InvalidLength)
+        }
+        let mut encoded = s.to_string();
+        encoded.insert(3, '1');
```

Comment:
What does this line do?

Reply:
This is the separator of Bech32. I had to decide whether to change Bech32 implementation or not, and I decided not to change because it's more simple.